File: packages/guards/src/emotional-content.ts
import type { Guard, GuardResult, GuardContext } from 'open-guardrail-core';

interface EmotionalContentOptions {
  action: 'block' | 'warn';
}

const PATTERNS: RegExp[] = [
  /act\s+now\s+(before|or)/i,
  /limited\s+time\s+(only|offer)/i,
  /don'?t\s+miss\s+(out|this)/i,
  /hurry\s+(up|before)/i,
  /last\s+chance/i,
  /you\s+(should\s+be|ought\s+to\s+feel)\s+(ashamed|guilty)/i,
  /if\s+you\s+really\s+cared/i,
  /only\s+a\s+(fool|idiot)\s+would/i,
  /you'?re\s+so\s+(smart|brilliant|amazing)/i,
  /everyone\s+is\s+(talking|outraged)\s+about/i,
  /this\s+will\s+(shock|blow\s+your\s+mind)/i,
  /you\s+won'?t\s+believe/i,
  /fear\s+of\s+missing\s+out/i,
  /FOMO/,
  /urgent\s*:?\s*action\s+required/i,
  /wake\s+up\s+sheeple/i,
  /they\s+don'?t\s+want\s+you\s+to\s+know/i,
];

export function emotionalContent(options: EmotionalContentOptions): Guard {
  return {
    name: 'emotional-content',
    version: '0.1.0',
    description: 'Detects emotionally charged content',
    category: 'content',
    supportedStages: ['output'],
    async check(text: string, _ctx: GuardContext): Promise<GuardResult> {
      const start = performance.now();
      const matched: string[] = [];

      for (const pattern of PATTERNS) {
        if (pattern.test(text)) {
          matched.push(pattern.source);
        }
      }

      const triggered = matched.length > 0;
      const score = triggered ? Math.min(matched.length / 4, 1.0) : 0;

      return {
        guardName: 'emotional-content',
        passed: !triggered,
        action: triggered ? options.action : 'allow',
        score,
        latencyMs: Math.round(performance.now() - start),
        details: triggered ? { matchedPatterns: matched.length } : undefined,
      };
    },
  };
}

File: python/open_guardrail/guards/emotional_content.py
"""Detect emotionally charged content."""
from __future__ import annotations

import re
import time

from open_guardrail.core import GuardResult

_PATTERNS: list[re.Pattern[str]] = [
    re.compile(
        r"act\s+now\s+(before|or)", re.IGNORECASE
    ),
    re.compile(
        r"limited\s+time\s+(only|offer)",
        re.IGNORECASE,
    ),
    re.compile(
        r"don'?t\s+miss\s+(out|this)", re.IGNORECASE
    ),
    re.compile(r"hurry\s+(up|before)", re.IGNORECASE),
    re.compile(r"last\s+chance", re.IGNORECASE),
    re.compile(
        r"you\s+(should\s+be|ought\s+to\s+feel)"
        r"\s+(ashamed|guilty)",
        re.IGNORECASE,
    ),
    re.compile(
        r"if\s+you\s+really\s+cared", re.IGNORECASE
    ),
    re.compile(
        r"only\s+a\s+(fool|idiot)\s+would",
        re.IGNORECASE,
    ),
    re.compile(
        r"you'?re\s+so\s+(smart|brilliant|amazing)",
        re.IGNORECASE,
    ),
    re.compile(
        r"everyone\s+is\s+(talking|outraged)"
        r"\s+about",
        re.IGNORECASE,
    ),
    re.compile(
        r"this\s+will\s+(shock|blow\s+your\s+mind)",
        re.IGNORECASE,
    ),
    re.compile(
        r"you\s+won'?t\s+believe", re.IGNORECASE
    ),
    re.compile(
        r"fear\s+of\s+missing\s+out", re.IGNORECASE
    ),
    re.compile(r"FOMO"),
    re.compile(
        r"urgent\s*:?\s*action\s+required",
        re.IGNORECASE,
    ),
    re.compile(
        r"wake\s+up\s+sheeple", re.IGNORECASE
    ),
    re.compile(
        r"they\s+don'?t\s+want\s+you\s+to\s+know",
        re.IGNORECASE,
    ),
]


class _EmotionalContent:
    def __init__(self, *, action: str = "warn") -> None:
        self.name = "emotional-content"
        self.action = action

    def check(
        self, text: str, stage: str = "output"
    ) -> GuardResult:
        start = time.perf_counter()
        matched: list[str] = []

        for pat in _PATTERNS:
            if pat.search(text):
                matched.append(pat.pattern)

        triggered = len(matched) > 0
        score = min(len(matched) / 4, 1.0) if triggered else 0.0
        elapsed = (time.perf_counter() - start) * 1000

        return GuardResult(
            guard_name="emotional-content",
            passed=not triggered,
            action=self.action if triggered else "allow",
            score=score,
            message=(
                "Emotional content detected"
                if triggered
                else None
            ),
            latency_ms=round(elapsed, 2),
            details=(
                {"matched_patterns": len(matched)}
                if triggered
                else None
            ),
        )


def emotional_content(
    *, action: str = "warn"
) -> _EmotionalContent:
    return _EmotionalContent(action=action)
